src/utils.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    indices_of_unique(x) -> Dict
+
+Return a `Dict` whose keys are the unique elements of `x` and whose values are the
+corresponding indices in `x`.
+"""
+function indices_of_unique(x)
+    d = Dict{eltype(x),Vector{Int}}()
+    for (i, xi) in enumerate(x)
```

Comment:
This will only give you the indices of `x` if `x` actually uses linear indices of type `Int`. But it seems it will fail for more general indices. Maybe one could use
```suggestion
    for i in eachindex(x)
        xi = x[i]
```
instead (and adapt the value type of `d` accordingly)?